cpp/src/arrow/sparse_tensor-test.cc

```diff
@@ -186,6 +186,58 @@ TEST(TestSparseCOOTensor, CreationFromTensor) {
   ASSERT_EQ(3, sidx->Value({11, 2}));
 }
 
+TEST(TestSparseCOOTensor, CreationFromNonContiguousTensor) {
+  std::vector<int64_t> shape = {2, 3, 4};
+  std::vector<int64_t> values = {1,  0, 0, 0, 2,  0, 0, 0, 0, 0, 3,  0, 0, 0, 4,  0,
+                                 5,  0, 0, 0, 6,  0, 0, 0, 0, 0, 11, 0, 0, 0, 12, 0,
+                                 13, 0, 0, 0, 14, 0, 0, 0, 0, 0, 15, 0, 0, 0, 16, 0};
+  std::vector<int64_t> strides = {192, 64, 16};
+  std::shared_ptr<Buffer> buffer = Buffer::Wrap(values);
+  Tensor tensor(int64(), buffer, shape, strides);
+  SparseTensorImpl<SparseCOOIndex> st(tensor);
+
+  ASSERT_EQ(12, st.non_zero_length());
+  ASSERT_TRUE(st.is_mutable());
+
+  const int64_t* ptr = reinterpret_cast<const int64_t*>(st.raw_data());
```

Comment:
This can of code occurs several time in the test file. Could you define a test helper for this (perhaps in `test-util.h`)? So this would become e.g.:
```cpp
AssertNumericDataEqual(st.raw_data(), std::vector<int64_t>{1, 2, 3, 4, 5, 6, 11, 12, 13, 14, 15, 16});
```

Reply:
Done.